dist/tools/drone-scripts/build_and_test.sh

```diff
@@ -19,27 +32,32 @@ export TRAVIS_BUILD_NUMBER="${CI_BUILD_NUMBER}"
 export TRAVIS_COMMIT="${DRONE_COMMIT}"
 export TRAVIS_BUILD_DIR="${DRONE_BUILD_DIR}"
 
-for group in static-tests \
-             avr8         \
-             msp430       \
-             x86          \
-             arm7         \
-             cortex_m0    \
-             cortex_m3_1  \
-             cortex_m3_2  \
-             cortex_m4    \
-    ;
-do
-    echo "Begin group ${group}"
-    BUILDTEST_MCU_GROUP=${group} ./dist/tools/travis-scripts/build_and_test.sh "$@"
-    RES=$?
-    echo "Result (${group}): ${RES}"
-    if [ ${RES} -ne 0 ]; then
-        FAILURES=$((${FAILURES} + 1))
-    fi
-    SUMMARY_MESSAGE="${SUMMARY_MESSAGE}${group}: ${RES}\n"
+export -f exec_build_func
+
+MYTMPDIR=$(mktemp -d)
+trap 'rm -rf "$MYTMPDIR"' EXIT
+
+# Execute all groups in parallel (-k ensures correct ordering of the
+# output)
+parallel -k exec_build_func {} "$@"  ::: static-tests avr8 msp430 x86 arm7 \
+                                         cortex_m0 cortex_m3_1 cortex_m3_2 \
+                                         cortex_m4 \
+|& tee -a "$MYTMPDIR/output.log"
+
+# Check the log for failures
+RESULTS=$(cat "$MYTMPDIR/output.log" | grep "Result.*:" | cut -d ":" -f2 \
+        | tr "\n" " ")
+
+for retval in $RESULTS;do
+	if [ $retval -ne "0" ];then
+		((FAILURES++))
+	fi
 done
 
+SUMMARY=$(cat "$MYTMPDIR/output.log" | grep -e "^Result.*:" \
+        | sed "s/Result\ (//" | sed "s/)//")
```

Comment:
The regex is correct in the sense that it produces the same output as your script used to do.
However, I have to admit that my solution is rather clumsy. Since, I couldn't  get your proposed solution to work I replaced sed with awk. This change makes this line (IMHO) much easier to understand.